### ISSUE TITLE
Add 'wp_body_open()' to Oxygen Main Template

### ISSUE THREAD

**Issue by wplit** [soflyy/oxygen-bugs-and-features#290]
from WP5.2+, theme devs are being told to add this function just after the opening body tag to add the new action hook, to standardise where users/plugins can hook in code that need to be added there in the body. (namely for GTM code, pixels..)

See [https://generatewp.com/wordpress-5-2-action-that-every-theme-should-use/](https://generatewp.com/wordpress-5-2-action-that-every-theme-should-use/)

By adding in Oxygen, it would make transferring sites from theme-to-Oxygen and Oxygen-to-theme smoother as the analytics plugins etc will continue to work without users having to think about a new place to put the code.  

I expect plugins such as [Google Tag Manager for WP](https://wordpress.org/plugins/duracelltomi-google-tag-manager/) will start using the hook soon, so this would mean Oxygen would automatically support that plugin and others like it once it does.

Currently we can use snippets or custom plugin to use the 'ct_before_builder' hook to put the code in the correct place (even if using the GTM for WP plugin, we still need to manually add) but this hook is obv Oxygen dependant, so doesn't help when switching from or to a theme. Much better if using the same hook as everybody else, like with wp_footer or wp_head.

**Comment by wplit** [comment]
`if ( function_exists( 'wp_body_open' ) ) {
	wp_body_open();
}`

**Comment by Oxymaniac** [comment]
+1

**Comment by dinolatoga** [comment]
+1 to this. Until now, this isn't still integrated. Please do add this on the next update.

**Comment by creative-prowess** [comment]
+1 basic Wordpress function.

**Comment by jevgen** [comment]
+1 Google Tags Manager already require this

**Comment by shmaltz** [comment]
Any update on this? Google Tag manager requires a tag be placed at the top of the <body> tag.

**Comment by TheLEAX** [comment]
+1

**Comment by StevenStavrakis** [comment]
Still not working...

**Comment by imsyedabdullah** [comment]
+1

**Comment by maltmann-muc** [comment]
+1

**Comment by perrelet** [comment]
How is this not a thing?

**Comment by perrelet** [comment]
The single line of code required to close this issue would have taken approximately the same amount of time to implement that it took @Spellhammer to remove the 'High Priority' tag.

![image](https://user-images.githubusercontent.com/7825147/131585630-0a334a43-6c0f-4dee-8e77-ba3ff37b2882.png)

So confused.

**Comment by hagsonz** [comment]
+1
meanwhile, you can add this to your Main template or create a parent template and inherit with Main..

```
if ( function_exists( 'wp_body_open' ) ) {
    wp_body_open();
} else {
    do_action( 'wp_body_open' );
}
```

**Comment by ricardopadilha** [comment]
Here's a ready-to-use patch to include the call, using @hagsonz's code:
```
--- a/component-framework/oxygen-main-template.php	2021-11-25 18:02:43.068251478 -0300
+++ b/component-framework/oxygen-main-template.php	2021-11-25 18:04:08.340553686 -0300
@@ -21,6 +21,13 @@
 <body <?php body_class($classes_list); ?> <?php if ( defined("SHOW_CT_BUILDER") ) : ?>id="ct-controller-ui" ng-controller="ControllerUI"<?php endif; ?><?php do_action("oxygen_vsb_body_attr"); ?>>

 <?php
+if (function_exists('wp_body_open')) {
+	wp_body_open();
+} else {
+	do_action('wp_body_open');
+}
+?>
+<?php
 	if ( defined("SHOW_CT_BUILDER") ) {
 		$is_404 = false;

```

Could we please have this merged as soon as possible?

**Comment by ricardopadilha** [comment]
Meanwhile, there is another possible workaround. This is a single line plugin that taps into an oxygen-specific hook, and that can be enabled until oxygen applies my previous patch:
```
<?php

/**
 * Plugin Name:       Oxygen wp_body_open
 * Description:       Uses ct_before_builder to call wp_body_open.
 * Version:           0.1
 * Requires at least: 5.2
 * License:           Public domain
 */

add_action('ct_before_builder', wp_body_open, 1);
```
This has been tested on Oxygen 3.8.1, and requires WordPress 5.2.

**Comment by ricardopadilha** [comment]
It looks like this was fixed in Oxygen 4.0. This is line 23 in plugins/oxygen/component-framework/oxygen-main-template.php:
```
<?php if ( !defined("SHOW_CT_BUILDER") ) wp_body_open(); ?>
```

This is not as robust as @hagsonz's suggestion, since [wp_body_open](https://developer.wordpress.org/reference/functions/wp_body_open/) has been introduced in WordPress 5.2.0, but I guess that anyone upgrading to Oxygen 4.0 is probably using a more recent version of WordPress than that.

In any case, the summary seems to be:
- If you are using Oxygen 3.x with an WordPress older than 5.2, then use the patch from [this comment](https://github.com/soflyy/oxygen-bugs-and-features/issues/290#issuecomment-979469556)
- If you are using Oxygen 3.x with WordPress 5.2 or greater, then use the one-line plugin from [this comment](https://github.com/soflyy/oxygen-bugs-and-features/issues/290#issuecomment-979479013)
- If you are using Oxygen 4.x, disable or remove the one-line plugin.

**Comment by KittenCodes** [comment]
Added in Oxygen v4.
